hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssMessageHandler.java

```diff
@@ -56,6 +71,44 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
         requireNonNull(context);
-        submissionManager.submitTssVote(TssVoteTransactionBody.DEFAULT, context);
+        final var op = context.body().tssMessageOrThrow();
+
+        final var tssStore = context.storeFactory().writableStore(WritableTssBaseStore.class);
+        final var rosterStore = context.storeFactory().readableStore(ReadableRosterStore.class);
+        final var maxSharesPerNode =
+                context.configuration().getConfigData(TssConfig.class).maxSharesPerNode();
+        final var numberOfAlreadyExistingMessages =
+                tssStore.getTssMessages(op.targetRosterHash()).size();
+
+        // The sequence number starts from 0 and increments by 1 for each new message.
+        final var key = TssMessageMapKey.newBuilder()
+                .rosterHash(op.targetRosterHash())
+                .sequenceNumber(numberOfAlreadyExistingMessages)
+                .build();
+        // Each tss message is stored in the tss message state and is sent to CryptographyManager for further
+        // processing.
+        tssStore.put(key, op);
+
+        final var tssParticipantDirectory =
+                computeTssParticipantDirectory(rosterStore.getActiveRoster(), maxSharesPerNode, (int)
+                        context.networkInfo().selfNodeInfo().nodeId());
+        final var result = tssCryptographyManager.handleTssMessageTransaction(op, tssParticipantDirectory, context);
+
+        result.thenAccept(ledgerIdAndSignature -> {
+            if (ledgerIdAndSignature != null) {
+                final var signature =
+                        gossip.sign(ledgerIdAndSignature.ledgerId().publicKey().toBytes());
+                // FUTURE: Validate the ledgerId computed is same as the current ledgerId
+                final var tssVote = TssVoteTransactionBody.newBuilder()
+                        .tssVote(Bytes.wrap(ledgerIdAndSignature.tssVoteBitSet().toByteArray()))
+                        .targetRosterHash(op.targetRosterHash())
+                        .sourceRosterHash(op.sourceRosterHash())
+                        .nodeSignature(signature.getBytes())
+                        .ledgerId(Bytes.wrap(
+                                ledgerIdAndSignature.ledgerId().publicKey().toBytes()))
+                        .build();
+                submissionManager.submitTssVote(tssVote, context);
+            }
```

Comment:
Would it be useful to log a warn/error message if `ledgerIdAndSignature` is null here, and that no TSS vote will be submitted? Not sure 🤔

Reply:
Its called for every TssMessageTransaction. Those will be too many logs imo.